pyro/ops/provenance.py

```diff
@@ -93,6 +93,11 @@ def _track_provenance_set(x, provenance: frozenset):
 @track_provenance.register(tuple)
 @track_provenance.register(dict)
 def _track_provenance_pytree(x, provenance: frozenset):
+    # avoid max-recursion depth error for torch<=2.0
+    flat_args, _ = tree_flatten(x)
+    if not flat_args or flat_args[0] is x:
+        return x
+
```

Comment:
This is a heuristic solution but we can always patch bug fixes if any bugs come up in the future. So I am happy with it.